src/celeritas/optical/CerenkovParams.cc

```diff
@@ -0,0 +1,62 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/CerenkovParams.cc
+//---------------------------------------------------------------------------//
+#include "CerenkovParams.hh"
+
+#include <utility>
+#include <vector>
+
+#include "corecel/cont/Range.hh"
+#include "corecel/data/CollectionBuilder.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/grid/GenericGridData.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with optical property data.
+ */
+CerenkovParams::CerenkovParams(HostCRef<OpticalPropertyData> const& properties)
+{
+    HostVal<CerenkovData> data;
+    for (auto mat_id :
+         range(OpticalMaterialId(properties.refractive_index.size())))
+    {
+        GenericGridData ai_grid;
+        auto const& ri_grid = properties.refractive_index[mat_id];
+        if (!ri_grid)
+        {
+            // No refractive index data stored for this material
+            make_builder(&data.angle_integral).push_back(ai_grid);
```

Comment:
Create a persistent builder for this and `reals` (outside the scope of the `for` loop)...